codegen/class-model/src/main/java/io/helidon/codegen/classmodel/ImportOrganizer.java

```diff
@@ -241,6 +238,17 @@ private void resolveFinalImports() {
             }
         }
 
+        private Type topLevelType(Type type) {
+            Type lastType = type;
+            Optional<Type> declaringClass = type.declaringClass();
+            while (declaringClass.isPresent()) {
+                lastType = declaringClass.get();
+                declaringClass = lastType.declaringClass();
+
+            }
```

Comment:
```suggestion
            if (declaringClass.isPresent()) {
                lastType = declaringClass.get();
                declaringClass = lastType.declaringClass();

            }
```

Reply:
As the comment on the call site suggests, we need the top level, not just the parent:
```
//If processed type is inner class, we will be importing (top level) parent class
```